### PR TITLE
Feat/ascat rules

### PR DESCRIPTION
**Pull request by khurrammaqbool**
### This PR:

(If this is a release PR, no need to add following. Leave this part empty)
(Use the following lines to create a PR text body. Make sure to remove all non-relevant one after you're done)
(Repeat each field as many times as necessary)

Added: ascatngs rules.

### Review and tests: 
- [ ] Tests pass
- [ ] Code review
- [ ] New code is executed and covered by tests, and test approve

### CODE REVIEW

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule [hunk 1]
```diff
@@ -81,13 +81,78 @@ rule delly_tumor_normal:
...
+        gccorrection = config["reference"]["ascat_gccorrection"],
+        chryloci= config["reference"]["ascat_chryloci"],
+    output:
+        finalvcf = vcf_dir + "SV.somatic." + config["analysis"]["case_id"] + ".ascat.vcf.gz",
+        samplestatistics = vcf_dir + "SV.somatic." + config["analysis"]["case_id"] + ".ascat.samplestatistics.txt",
+        germlineplot = vcf_dir + "SV.somatic." + config["analysis"]["case_id"] + ".ascat.germline.png",
+        rawprofileplot = vcf_dir + "SV.somatic." + config["analysis"]["case_id"] + ".ascat.rawprofile.png",
+        sunriseplot = vcf_dir + "SV.somatic." + config["analysis"]["case_id"] + ".ascat.sunrise.png",
+        tumorplot = vcf_dir + "SV.somatic." + config["analysis"]["case_id"] + ".ascat.tumor.png",
+        ASPCFplot = vcf_dir + "SV.somatic." + config["analysis"]["case_id"] + ".ascat.ASPCF.png",
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule:115]: lower case please if possible.

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule [hunk 2]
```diff
@@ -81,13 +81,78 @@ rule delly_tumor_normal:
...
+    singularity:
+        Path(singularity_image, config["bioinfo_tools"].get("ascat") + ".sif").as_posix()
+    params:
+        tmpdir = tempfile.mkdtemp(prefix=tmp_dir),
+        runmode = "local",
+        tumor = get_sample_type(config["samples"], "tumor"),
+        normal = get_sample_type(config["samples"], "normal"),
+        conda = config["bioinfo_tools"].get("ascat")
+    threads:
+        get_threads(cluster_config, "ascat_tumor_normal")
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule:129]: add this to https://github.com/Clinical-Genomics/BALSAMIC/blob/master/BALSAMIC/config/cluster.json file

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule [hunk 3]
```diff
@@ -81,13 +81,78 @@ rule delly_tumor_normal:
...
+    message:
+        ("Call copy number variants using ascatNGS for {input.bamT} vs {input.bamN} files then "
+        "filter somatic variants and finally convert to compressed vcf file")
+    shell:
+        """
+source activate {params.conda};
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule:135]: There is no need for conda in ascat container

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule [hunk 4]
```diff
@@ -81,13 +81,78 @@ rule delly_tumor_normal:
...
+    benchmark:
+        benchmark_dir + 'ascat_tumor_normal_' + config["analysis"]["case_id"] + "_ascat.tsv"
+    singularity:
+        Path(singularity_image, config["bioinfo_tools"].get("ascat") + ".sif").as_posix()
+    params:
+        tmpdir = tempfile.mkdtemp(prefix=tmp_dir),
+        runmode = "local",
+        tumor = get_sample_type(config["samples"], "tumor"),
+        normal = get_sample_type(config["samples"], "normal"),
+        conda = config["bioinfo_tools"].get("ascat")
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule:127]: there is no need for conda in ascat container.

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule [hunk 5]
```diff
@@ -81,13 +81,78 @@ rule delly_tumor_normal:
...
+    message:
+        ("Call copy number variants using ascatNGS for {input.bamT} vs {input.bamN} files then "
+        "filter somatic variants and finally convert to compressed vcf file")
+    shell:
+        """
+source activate {params.conda};
+
+export LD_LIBRARY_PATH=:/opt/wtsi-cgp/lib;
+
+ascat.pl -protocol WGS -species human -assembly GRCh37 -cpus 36 -reference {input.fa} -snp_gc {input.gccorrection} -outdir {params.tmpdir} -gender L -locus {input.chryloci} -tumour {input.bamT} -normal {input.bamN};
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_normal.rule:139]: Is it possible to have this as multiline command please?

File: BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_only.rule [hunk 6]
```diff
@@ -74,15 +74,78 @@ rule delly_tumor_only:
...
 tabix -p vcf -f {output.final};
 
 echo -e \"{params.tumor}\\tTUMOR\" > {output.namemap};
 
 rm -rf {params.tmpdir};
         """
+
+
+
+rule ascat_tumor_only:
```
Comment (hassanfa) [inline BALSAMIC/snakemake_rules/variant_calling/somatic_sv_tumor_only.rule:92]: similar comment to tumor_normal apply here as well. so I won't repeat them.

File: CHANGELOG.rst [hunk 7]
```diff
@@ -21,7 +21,7 @@ Added:
 * Reference section to docs/FAQ.rst
 * ascat download references
 * Delly tumor only rule
-
+* ascat tumor normal and tumor only rules
```
Comment (hassanfa) [inline CHANGELOG.rst:24]:
```suggestion
* ascat tumor normal and tumor only rules to WGS workflow
```

**Review by hassanfa — CHANGES_REQUESTED** [review]
Some general comments.

Please paste an screenshot that show ascatngs in the final workflow job names.

**Comment by khurrammaqbool** [comment]
Fixed in https://github.com/Clinical-Genomics/BALSAMIC/pull/699

**Pull request closed without merging**